presale/models/presale_model.py

```diff
@@ -0,0 +1,66 @@
+from odoo import fields, models
+
+
+class PresaleModel(models.Model):
+    _name = "presale.order"
+    _description = "Presale order model"
+
+    def _defaultName(self):
+        # TODO: Avoid searching the whole table
+        number = 1
+        record_names = [record.name for record in self.env["presale.order"].search([])]
+        while ("PS " + str(number).zfill(3)) in record_names:
+            number += 1
+        return "PS " + str(number).zfill(3)
+
+    active = fields.Boolean(default=True)
+    name = fields.Char(default=_defaultName, required=True)
+    partner_id = fields.Many2one("res.partner", string="Customer")
+    state = fields.Selection(
+        default="new",
+        required=True,
+        selection=[("new", "Draft"), ("validated", "Validated"), ("cancelled", "Cancelled")],
+    )
+    order_line_ids = fields.One2many("presale.order.line", "presale_order_id")
+
+    def send_validation_email(self, presale_order):
```

Comment:
Here you can use a mail template instead of creating the body directly in python.
That way the mail template can be modified more easily !
if you want i can give you an example of mail template i made for this training